From 5a369c3b3a835019e3bc82d9da6331c6fb50b40e Mon Sep 17 00:00:00 2001
From: Jan Wyszynski <jan.wyszynski@joyent.com>
Date: Tue, 11 Sep 2018 02:06:47 +0000
Subject: [PATCH] MANTA-3941 garbage-collector transformer needs to ignore
 requests to flush empty shark list Reviewed by: Jared Morrow <jm@joyent.com>
 Approved by: Jared Morrow <jm@joyent.com>

---
 lib/delete_record_transformer.js | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/delete_record_transformer.js b/lib/delete_record_transformer.js
index 2d6987f..eeebed6 100644
--- a/lib/delete_record_transformer.js
+++ b/lib/delete_record_transformer.js
@@ -358,6 +358,10 @@ _listen_for_records()
 				self.mt_moray_listener.emit('cleanup', key);
 				return;
 			}
+			if (sharks.length === 0) {
+				self.mt_log.info('Skipping no-op flush.');
+				return;
+			}
 			self._flush(sharks, function (ferr) {
 				if (ferr) {
 					self.mt_log.warn({
-- 
2.21.0

